### PR TITLE
C_CPP: Destructor in CPP

### LINKED ISSUE #3357: Destructors in CPP
### Description

Description

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by RahulRam29**
<hr>

## Description 📜

I have explained about the basic concepts of destructor along with it's syntax. For better understanding I have performed a program and attached it's output. 

Fixes #3357

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

<hr>

### CODE REVIEW

File: C_CPP/OOPs in C++/Constructor & Destructor/Advanced-Destructor.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# Destructor Concepts
+
+#### What is a destructor ?
+  Destructors are basically used for the **deallocation of memory when object goes out of scope** and it is used for class object and it's class members. Destructor has the same name as that of class name but it's preceded with ~(tilde).     There can be only one destructor in a class.    
+  There should be no return type and no parameters.
+
+ ### Syntax for using Destructor:
+ Suppose if class name is Add(),
```
Comment (deeqakkk) [inline C_CPP/OOPs in C++/Constructor & Destructor/Advanced-Destructor.md:8]: update it with " Suppose if class name is Add(),
 then ~add()  will deallocate the memory "

**Review by deeqakkk — CHANGES_REQUESTED** [review]
Rahul, your content is nice. But it doesn't answer "what, Why & how" clearly, so add more detailing and concepts.

File: C_CPP/OOPs in C++/Constructor & Destructor/Advanced-Destructor.md (hunk 1 above)
Comment (RahulRam29) [reply C_CPP/OOPs in C++/Constructor & Destructor/Advanced-Destructor.md]: I have made the changes that you have asked me to change and saved it sir.
Thank you sir.

**Review by kashishahuja2002 — CHANGES_REQUESTED** [review]
Great content, but the code is quite confusing for the topic of destructor. Try adding a simpler code snippet.

**Comment by RahulRam29** [comment]
I have added a simple code snippet as you have advised me mam.
Thank you mam.

**Review by kashishahuja2002 — CHANGES_REQUESTED** [review]
For this create a pull request on another branch.

**Comment by RahulRam29** [comment]
@kashishahuja2002  Check if it's fine now.

**Review by kashishahuja2002 — CHANGES_REQUESTED** [review]
There should be only one file. Merge the contents of "C_CPP/OOPs in C++/Constructor & Destructor/readme.md" file into "C_CPP/OOPs in C++/Constructor & Destructor/Advanced-Destructor.md" file.

**Comment by RahulRam29** [comment]
@kashishahuja2002 I understand what you are saying, but Advanced-Destructor.md is a file inside the 'Constructor & Destructor' directory. So, it is not possible to merge readme.md file with Advanced-Destructor.md file. Can you refer it once again please?

**Comment by Saviour1001** [comment]
Looks great to me !

**Review by kashishahuja2002 — COMMENTED** [review]
Apologies, I didn't see it correctly, I thought you added a different readme in the file.

**Pull request merged**